src/python_testing/TC_ICDM_3_3.py

```diff
@@ -0,0 +1,380 @@
+            try:
+                await self._send_single_icdm_command(commands.UnregisterClient(checkInNodeID=client6.checkInNodeID, verificationKey=kIncorrectKey))
+            except InteractionModelError as e:
+                asserts.assert_equal(
+                    e.status, Status.Success, "Unexpected error returned")
+
+            self.step("6d")
+            registeredClients = await self._read_icdm_attribute_expect_success(
+                attributes.RegisteredClients)
+            asserts.assert_equal(len(registeredClients), 0,
+                                 "The RegisteredClients list must be empty. List has the wrong size.")
+            self.step(7)
+            ac = Clusters.AccessControl
+            previousAcl = await self.read_single_attribute_check_success(cluster=ac, attribute=ac.Attributes.Acl)
+            newAcls = []
+
+            # Set Admin permissions on Access Control cluster
+            newAclEntry = ac.Structs.AccessControlEntryStruct(privilege=ac.Enums.AccessControlEntryPrivilegeEnum.kAdminister,
+                                                              authMode=ac.Enums.AccessControlEntryAuthModeEnum.kCase,
+                                                              subjects=previousAcl[0].subjects, targets=[ac.Structs.AccessControlTargetStruct(
+                                                                  cluster=Clusters.AccessControl.id)], fabricIndex=previousAcl[0].fabricIndex
+                                                              )
+            newAcls.append(newAclEntry)
+
+            # Set Manage permissions on ICD Management cluster
+            newAclEntry = ac.Structs.AccessControlEntryStruct(privilege=ac.Enums.AccessControlEntryPrivilegeEnum.kManage,
+                                                              authMode=ac.Enums.AccessControlEntryAuthModeEnum.kCase,
+                                                              subjects=previousAcl[0].subjects, targets=[ac.Structs.AccessControlTargetStruct(
+                                                                  cluster=Clusters.IcdManagement.id)], fabricIndex=previousAcl[0].fabricIndex
+                                                              )
+            newAcls.append(newAclEntry)
+
+            try:
+                await self.default_controller.WriteAttribute(nodeid=self.dut_node_id, attributes=[(0, ac.Attributes.Acl(newAcls))])
+            except InteractionModelError as e:
+                asserts.assert_equal(
+                    e.status, Status.Success, "Unexpected error returned")
+
+            self.step("8a")
+            try:
+                await self._send_single_icdm_command(commands.RegisterClient(checkInNodeID=client8.checkInNodeID, monitoredSubject=client8.subjectId, key=client8.key, clientType=client8.clientType))
+            except InteractionModelError as e:
+                asserts.assert_equal(
+                    e.status, Status.Success, "Unexpected error returned")
```

Comment:
Here the test plan states:

> {resDutSuccess}.
{DUTreply} {A_ICD_COUNTER}.

Which leads me to believe there should be some other check, or otherwise, aren't the `{DUTreply} {A_ICD_COUNTER}` bits in the test plan redundant?